include/onnxruntime/core/common/optional.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/common/common.h"
+
+namespace onnxruntime {
+// This is a version of std::optional with limited functionality and plenty of
+// room to improve. We should use std::optional when we move to C++17.
+template <typename T>
```

Comment:
Would using boost::optional be better? [https://github.com/boostorg/optional](https://github.com/boostorg/optional)

Reply:
not sure, but we can look into it in future release as replacement.